packages/cli-repl/src/cli-repl.ts

```diff
@@ -44,17 +44,37 @@ class CliRepl {
   private mongoshDir: string;
 
   /**
-   * Connect to the cluster.
+   * Instantiate the new CLI Repl.
+   */
+  constructor(driverUri: string, driverOptions: NodeOptions, options: CliOptions) {
+    this.options = options;
+    this.mongoshDir = path.join(os.homedir(), '.mongodb/mongosh/');
+
+    this.createMongoshDir();
+
+    this.bus = new Nanobus('mongosh');
+    logger(this.bus, this.mongoshDir);
+
+    this.generateOrReadTelemetryConfig();
+
+    if (this.isPasswordMissing(driverOptions)) {
+      this.requirePassword(driverUri, driverOptions);
+    } else {
+      this.setupRepl(driverUri, driverOptions);
```

Comment:
technically this one has still the potential to generate an unhandled rejection, i see you've got a try catch around the body of the internal calls to connect, getBuild etc.., but to be really sure we should add a catch here:

```
this.setupRepl(driverUri, driverOptions).catch((error) => {
  // ...
   console.log(formatError(error))
});
```

Reply:
I am not super familiar with this syntax: does doing a `.catch()` here mean we don't have to wrap internal `await` functions in a `try`/`catch`?

Reply:
yep, it acts on the promise itself.

`catch` is a method of the promise that is used to setup a callback for when the promise is rejected;

```
async function fn() {
  throw new Error('err')
}

fn().catch((error) => {
  console.error('Got error:', error);
});
```

Reply:
cool, yea that's what I did. All the `await` methods are getting handled in the same way, so this saves me a bunch of lines \o/